src/verifySignedMessage.js

```diff
@@ -0,0 +1,130 @@
+const recoverSigner = require('./recoverSigner')
+const proxyAccountFromOwner = require('./proxyAccountFromOwner')
+const { toChecksumAddress, isAddress } = require('web3-utils')
+
+const ethToTokenParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'token', type: 'address'},
+  { name: 'ethAmount', type: 'uint256'},
+  { name: 'tokenAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const tokenToEthParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'token', type: 'address'},
+  { name: 'tokenAmount', type: 'uint256'},
+  { name: 'ethAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const tokenToTokenParamTypes = [
+  { name: 'bitmapIndex', type: 'uint256' },
+  { name: 'bit', type: 'uint256'},
+  { name: 'tokenIn', type: 'address'},
+  { name: 'tokenOut', type: 'address'},
+  { name: 'tokenInAmount', type: 'uint256'},
+  { name: 'tokenOutAmount', type: 'uint256'},
+  { name: 'expiryBlock', type: 'uint256'},
+  { name: 'to', type: 'address'},
+  { name: 'data', type: 'bytes'}
+]
+
+const verifySignedMessage = (signedMessage) => {
+  const { accountAddress, signature, message, signer } = signedMessage
+
+  if (!accountAddress) {
+    throw new Error('verifySignedMessage Error: Account Address not provided in signedMessage')
+  }
+
+  if (!signature) {
+    throw new Error('verifySignedMessage Error: Signature not provided in signedMessage')
+  }
+
+  if (!message) {
+    throw new Error('verifySignedMessage Error: Message not provided in signedMessage')
+  }
+
+  if (!signer) {
+    throw new Error('verifySignedMessage Error: Signer not provided in signedMessage')
+  }
+
+  const recoveredSigner = recoverSigner({ signature, typedDataHash: message})
+  if (recoveredSigner !== signer) {
+    throw new Error(`verifySignedMessage Error: Provided Signer ${signer} does not match Signer ${recoveredSigner} in Signed Message`)
+  }
+
+  const computedAccountAddress = accountFromOwner(signer)
+  if (toChecksumAddress(computedAccountAddress) !== toChecksumAddress(accountAddress)) {
+    throw new Error(`verifySignedMessage Error: Account Address ${accountAddress} does not match Computed Address ${computedAccountAddress}`)
+  }
+
+  const signedParams = signedMessage.signedParams
+  if (!signedParams || !signedParams[1]) {
+    throw new Error('verifySignedMessage Error: signedParams not provided in signedMessage')
+  }
+
```

Comment:
one more thing to verify here ... `signedMessage.functionName` has to be one of the supported meta fns in Account.sol. I think it has to be the string `metaDelegateCall` or `metaDelegateCall_EIP1271`, otherwise it won't work with a Brink account

Reply:
actually just needs to match `metaDelegateCall` since there is no offline signed msg for the 1271 call